winml/lib/Api/LearningModelSession.cpp

```diff
@@ -114,6 +114,10 @@ void LearningModelSession::Initialize() {
   if (session_options_ && session_options_.BatchSizeOverride() != 0) {
     WINML_THROW_IF_FAILED(engine_builder->SetBatchSizeOverride(session_options_.BatchSizeOverride()));
   }
+  com_ptr<winmlp::LearningModelSessionOptions> session_options_impl = session_options_.as<winmlp::LearningModelSessionOptions>();
+  if (session_options_ && session_options_impl->NamedDimensionOverrides().size() > 0) {
```

Comment:
Did you mean `session_options_impl && session_options_impl->NamedDimensionOverrides().size() > 0` ?

Reply:
And I think clang-tidy complains that you should use `.empty()` instead of comparing `.size()` to zero :) Doesn't make much of a difference for map, but in other containers computing the size might not be O(1)

Reply:
yeah, I should restructure this check to first check the session_options and then create the session_options_impl

Reply:
Updated!